releasenotes/notes/handle-deferred-close-forwarder.yaml

```diff
@@ -0,0 +1,5 @@
+---
+security:
+  - |
+    Fixes cwe 703. Handle the return value of Close() for writtable files and forces writes to disks
```

Comment:
> and forces writes to disks

Handling errors from Close is good, but how it forces anything? We only now handle the error if the write failed, but if the write was going to succeed, it still would.

Also, perhaps we don't need to mention CWE here.

Reply:
To explain my reasoning, `Close()` uses the `close` syscall which can return 3 errors:
- [EBADF]           fildes is not a valid, active file descriptor.
- [EINTR]            Its execution was interrupted by a signal.
- [EIO]                A previously-uncommitted write(2) encountered an input/output error.

We are interested in the latest. Since this error concerns uncommitted writes which encountered an I/O error, perhaps writes that do not encounter an error are committed by `Close()`.  This is my own understanding of the article but it is possible that I am wrong.

Reply:
> perhaps writes that do not encounter an error are committed by Close()

Right, but we were closing the file before this PR too. So as far as changelog is concerned, this "commit pending writes" behavior did not change.

Reply:
> A way to enforce that writes are committed is to use f.Sync() before closing the file but it might hurt performances.

And we don't need to. Even though this code uses word "transactions", it is not the same transactions as one finds in databases. The durability requirements are different.